source/server-discovery-and-monitoring/server-monitoring.rst

```diff
@@ -190,11 +190,6 @@ Each time a check completes, threads waiting for a `suitable`_ server
 are unblocked. Each unblocked thread MUST proceed if the new TopologyDescription
 now contains a suitable server.
 
-As an optimization, the client MAY leave threads blocked
-if a check completes without detecting any change besides
-roundTripTime: no operation that was blocked will
-be able to proceed anyway.
```

Comment:
This optimization could still be implemented, it would just need to ensure that even when the threads are blocked, an immediate check is still requested. That being said I'm fine with removing it entirely. I'm not sure any driver ever implemented this optimization.